### ISSUE TITLE
Event Binding Not Working in 1.2 Preview

### ISSUE THREAD

**Issue by brandonhall** [watusi/jquery-mobile-iscrollview#28]
Hey Jon,

I'm having trouble binding the events to listen for scroll changes.

Here's the setup:

```
$.mobile.iscrollview.prototype.options.adaptPage = false;
$.mobile.iscrollview.prototype.options.bounce = false;
$.mobile.iscrollview.prototype.options.createScroller = false;
$.mobile.iscrollview.prototype.options.hScroll = true;
$.mobile.iscrollview.prototype.options.hScrollbar = true;
$.mobile.iscrollview.prototype.options.lockDirection = false;
$.mobile.iscrollview.prototype.options.resizeWrapper = false;
$.mobile.iscrollview.prototype.options.preventPageScroll = false;
$.mobile.iscrollview.prototype.options.zoom = true;

this._scroller = this.$el.find(':jqmData(iscroll)').iscrollview();

```

Everything works fine and performs well, but when I try to intercept an event using:

```
$(document).delegate('.scrollWrapper', 'iscroll_onbeforescrollmove', function(event, data) {
    console.log('event');
    debugger;
});
```

... the callback is never fired.

`scrollWrapper` is the element that has the `data-iscroll` attribute on it. I've tried attaching the event to other elements without any luck either.

I'm seeing the event fired within the `jquery.mobile.iscrollview.js` file but it's never passed back to my callback. 

The only reason I'm trying to intercept the event is so we can tell the parent page to scroll when you reach the end of the iscroll container. We have the scroller embedded within a page that contains other usable elements. Do you know an easy way to achieve this behavior out of the box?

Thanks :)

**Comment by jtara** [comment]
pull-example.js in the demo binds to `iscroll_onpullup` and `iscroll_onpulldown`, so the mechanism works in general.

I will take a look to see if there might be some specific issue with `iscroll_onbeforescrollmove`.

**Comment by brandonhall** [comment]
I've tried it with several events (refresh, scrollMove) and haven't had any luck. Which element should it be attached too?

Here's our HTML:

```
            <div class="scrollWrapper" data-iscroll>
                <div class="scrollViewport" style="width:<@=width@>px;">
                    <div class="scroll_11949494">
                        <div style="z-index:0;">
                        <div style="z-index:-9999;background-color:#FFF;">
                            <div class="textCover"></div>
                            <div class="text"></div>
                        </div>
                        </div>
                        <div style="clear:both;"></div>
                    </div>
                </div>
            </div>
```

The scrollViewport is the element actually scrolled. Would that even make a difference in this instance?

**Comment by jtara** [comment]
Fixed in commit 8c037797

Callbacks for all of the reflected iScroll events were broken. Only the widget-specific events (like iscroll_onPullDown) were working. Doh!

Thanks!

(It's a bitch moving between Javascript and Ruby!)

Oh - to answer your question: for all events supported by iscrollview, you bind to the wrapper element.

**Comment by brandonhall** [comment]
Great thanks for fixing it!
